Prepare for Ecto 3.11

greg-rychlewski opened the pull request:
- Support fragment splicing, closes https://github.com/elixir-sqlite/ecto_sqlite3/issues/117
- Support parent_as with combination queries, reference: https://github.com/elixir-ecto/ecto_sql/pull/561
- Don't need to consider `{:maybe, type}` when loading, dumping. Closes: https://github.com/elixir-sqlite/ecto_sqlite3/issues/120
- Handle `nil` values in dumpers/loaders, reference: https://github.com/elixir-ecto/ecto/pull/4214.

I put everything into this PR so it can be merged once you use Ecto 3.11.  But you could pick out the parent_as one and commit it now (minus the new test) if you prefer

greg-rychlewski commented:
It looks like some new integration tests are failing. I'll take a look.

greg-rychlewski commented:
Ok I found the last culprit. Previously Ecto would not send `nil` down to the adapter but it was requested to send it through here https://github.com/elixir-ecto/ecto/pull/4214.

So this means your bool_encode function would have to take `nil` into account.

warmwaffles commented:
Is there a timeline for Ecto 3.11 release?

greg-rychlewski commented:
Not at the moment. There is someone working on a Cassandra adapter that might ask for a release once they are ready. But other than that no real indication of when it will happen.

I don't mind keeping this up to date with the changes that happen until then.

warmwaffles commented:
I'll keep an eye out for the 3.11 release. When that lands I'll ping you and we should be able to get this squared away and released.

greg-rychlewski commented:
Ecto 3.11 was released yesterday. I updated the deps and also I added a clause to raise if someone tries to use values lists with this adapter. When I looked up how values lists are handled in sqlite it was a bit weird because you can't give column names. I'm not too sure the best way to handle it atm.

warmwaffles commented:
Interesting, someone made a PR asking for a callback before each disconnect as well https://github.com/elixir-sqlite/exqlite/blob/5b9891bbe7d965532de6e3e8ca5c82735f864158/lib/exqlite/connection.ex#L162-L164
